raster/src/main/scala/geotrellis/raster/io/geotiff/CompressedBytesArray.scala

```diff
@@ -0,0 +1,71 @@
+package geotrellis.raster.io.geotiff
+
+import geotrellis.vector.Extent
+import geotrellis.raster._
+import geotrellis.raster.io.geotiff._
+import geotrellis.raster.io.geotiff.reader._
+import geotrellis.raster.io.geotiff.tags._
+import geotrellis.raster.io.geotiff.util._
+
+import java.nio.ByteBuffer
+import monocle.syntax.apply._
+import spire.syntax.cfor._
+
+class CompressedBytesArray(compressedBytes: Array[Array[Byte]]) extends CompressedBytes {
```

Comment:
Naming convention goes from left being the most specific, right being the most general.
Not everything follows this convention in the base library, but we should follow it consistently in this library (and I think we generally do with a couple of exceptions perhaps).

`ArrayCompressedBytes`
`ByteBufferCompressedBytes`

would fit better.

Also perhaps a different name would be more descriptive.

`SegmentBytes`
`ArraySegmentBytes`
`BufferSegmentBytes`

Thoughts?